rpcs3/Emu/RSX/GL/rsx_gl_texture.cpp

```diff
@@ -138,7 +137,22 @@ namespace rsx
 				glPixelStorei(GL_UNPACK_SWAP_BYTES, GL_TRUE);
 				glPixelStorei(GL_UNPACK_ROW_LENGTH, pitch / 2);
 
-				// TODO: texture swizzling
+				if (is_swizzled)
+				{
+					u16 height = tex.height();
```

Comment:
This whole block is almost the same as the new block below, why not refactor it into its own function?